### PR TITLE
[bn] Localize concepts/extend-kubernetes/_index.md

### PR DESCRIPTION
**Pull request by asem-hamid**
Localize concepts/extend-kubernetes/_index.md in bangla

Fixes Issue : https://github.com/kubernetes/website/issues/44993
Related PR : https://github.com/kubernetes/website/pull/45009
Tracking Issue: https://github.com/kubernetes/website/issues/31677

### CODE REVIEW

File: content/bn/docs/concepts/extend-kubernetes/_index.md [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+ক্লাস্টারের সাথে কাজ করে।
+
+ক্লায়েন্ট প্রোগ্রাম লেখার জন্য একটি নির্দিষ্ট প্যাটার্ন রয়েছে যা কুবারনেটিসের
+সাথে ভালভাবে কাজ করে যাকে {{< glossary_tooltip term_id="controller" text="কন্ট্রোলার" >}} 
+প্যাটার্ন বলা হয়। কন্ট্রোলাররা সাধারণত একটি অবজেক্টের `.spec` পড়ে, সম্ভবত জিনিসগুলি করে এবং 
+তারপর অবজেক্টের `.status` আপডেট করে ৷
+
+একটি কন্ট্রোলার হল কুবারনেটিস API-এর ক্লায়েন্ট। যখন কুবারনেটিস ক্লায়েন্ট হয় এবং 
+একটি রিমোট সার্ভিসে কল করে, কুবারনেটিস এটিকে একটি *webhook* বলে। রিমোট সার্ভিসকে 
+*webhook backend* বলা হয়। কাস্টম কন্ট্রোলারের মতো, webhookগুলো ব্যর্থতার একটি পয়েন্ট যোগ করে।
```
Comment (sajjadrahman56) [inline content/bn/docs/concepts/extend-kubernetes/_index.md:84]: webhook গুলো

File: content/bn/docs/concepts/extend-kubernetes/_index.md [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+
+{{< note >}}
+কুবারনেটিসের বাইরে, “webhook” শব্দটি সাধারণত অ্যাসিঙ্ক্রোনাস(asynchronous) বিজ্ঞপ্তিগুলির জন্য একটি প্রক্রিয়াকে বোঝায়, 
+যেখানে webhook কল অন্য সিস্টেম বা উপাদানের জন্য একমুখী বিজ্ঞপ্তি হিসাবে কাজ করে।
+কুবারনেটিস ইকোসিস্টেমে, এমনকি সিঙ্ক্রোনাস(synchronous) HTTP কলআউটগুলোকে প্রায়শই
```
Comment (sajjadrahman56) [inline content/bn/docs/concepts/extend-kubernetes/_index.md:89]: প্রায়শই ->  প্রায়ই

File: content/bn/docs/concepts/extend-kubernetes/_index.md [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+   `pods`, কুবারনেটিস প্রজেক্ট দ্বারা সংজ্ঞায়িত করা হয় এবং পরিবর্তন করা যায় না।
+   কুবারনেটিস API প্রসারিত করার বিষয়ে জানতে [API এক্সটেনশন](#API-এক্সটেনশন) পড়ুন।
+
+1. কুবারনেটিস শিডিউলার কোন নোডগুলোতে পড স্থাপন করবে তা
+   [নির্ধারণ](/bn/docs/concepts/scheduling-eviction/assign-pod-node/) করে। শিডিউলিং প্রসারিত করার
+   বিভিন্ন উপায় রয়েছে, যা [শিডিউলিং এক্সটেনশন](#শিডিউলিং-এক্সটেনশন) বিভাগে বর্ণিত করা হয়েছে।
+
+1. কুবারনেটিসের বেশিরভাগ আচরণ {{< glossary_tooltip term_id="controller" text="কন্ট্রোলার" >}}
+   নামক প্রোগ্রাম দ্বারা বাস্তবায়িত হয়, যেগুলো API সার্ভারের ক্লায়েন্ট।
+   কন্ট্রোলারগুলো প্রায়শই কাস্টম রিসোর্সগুলোর সাথে একত্রে ব্যবহৃত হয়।
```
Comment (sajjadrahman56) [inline content/bn/docs/concepts/extend-kubernetes/_index.md:130]: প্রায়ই

File: content/bn/docs/concepts/extend-kubernetes/_index.md [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+কাস্টম কোডে কল করার অনুমতি দেয় যা একটি অনুমোদনের সিদ্ধান্ত নেয়।
+
+### ডাইনামিক অ্যাডমিশন কন্ট্রোল
+
+একটি অনুরোধ অনুমোদিত হওয়ার পরে, যদি এটি একটি লিখিত অপারেশন হয়, তবে এটি 
+[অ্যাডমিশন কন্ট্রোলের](/bn/docs/reference/access-authn-authz/admission-controllers/) পদক্ষেপগুলোর
+মধ্য দিয়ে যায়। বিল্ট-ইন পদক্ষেপগুলো ছাড়াও, বেশ কয়েকটি এক্সটেনশন রয়েছে:
+
+* [ইমেজ পলিসি webhook](/bn/docs/reference/access-authn-authz/admission-controllers/#imagepolicywebhook)
+  কন্টেইনারে কোন ছবি চালানো যাবে তা সীমাবদ্ধ করে।
```
Comment (sajjadrahman56) [inline content/bn/docs/concepts/extend-kubernetes/_index.md:241]: ছবি  -> ইমেজ

File: content/bn/docs/concepts/extend-kubernetes/_index.md [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+* ইচ্ছামত অ্যাডমিশন কন্ট্রোলের সিদ্ধান্ত নিতে, একটি সাধারণ
+  [অ্যাডমিশন webhook](/bn/docs/reference/access-authn-authz/extensible-admission-controllers/#admission-webhooks)
+  ব্যবহার করা যেতে পারে। অ্যাডমিশন ওয়েবহুক সৃষ্টি বা আপডেট প্রত্যাখ্যান করতে পারে।
```
Comment (sajjadrahman56) [inline content/bn/docs/concepts/extend-kubernetes/_index.md:244]: ওয়েবহুক(webhook) | webhook

File: content/bn/docs/concepts/extend-kubernetes/_index.md [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+  কিছু অ্যাডমিশন webhook ইনকামিং রিকোয়েস্ট ডেটা পরিবর্তন করে কুবারনেটিস দ্বারা আরও পরিচালনা করার আগে।
+
+## অবকাঠামো এক্সটেনশন
+
+### ডিভাইস প্লাগইন
+
+_ডিভাইস প্লাগইনগুলো_ একটি [ডিভাইস প্লাগইনের](/bn/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/) 
+মাধ্যমে একটি নোডকে নতুন নোড রিসোর্সগুলো (সিপিইউ এবং মেমরির মতো বিল্টইনগুলো ছাড়াও)
```
Comment (sajjadrahman56) [inline content/bn/docs/concepts/extend-kubernetes/_index.md:252]: সিপিইউ -> CPU

File: content/bn/docs/concepts/extend-kubernetes/_index.md [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+
+### স্টোরেজ প্লাগইন
+
+{{< glossary_tooltip text="কন্টেইনার স্টোরেজ ইন্টারফেস(Container Storage Interface)" term_id="csi" >}} (CSI) 
+প্লাগইনগুলো নতুন ধরনের ভলিউমের জন্য সাপোর্ট সহ কুবারনেটিস প্রসারিত করার একটি উপায় প্রদান করে।
+ভলিউমগুলো টেকসই এক্সটার্নাল স্টোরেজ দ্বারা সাহায্যপ্রাপ্ত করা যেতে পারে, বা ক্ষণস্থায়ী স্টোরেজ(ephemeral storage) প্রদান করতে পারে, 
+অথবা তারা একটি ফাইল সিস্টেম প্যারাডাইম ব্যবহার করে তথ্যের জন্য একটি রিড-অনলি ইন্টারফেস দিতে পারে ।
+
+কুবারনেটিস এছাড়াও FlexVolume প্লাগইনগুলোর জন্য সাপোর্ট অন্তর্ভুক্ত করে, 
+যা কুবারনেটিস v1.23 (CSI-এর পক্ষে) থেকে ডেপ্রিসিয়েটেড(deprecated) করা হয়েছে ।
```
Comment (sajjadrahman56) [inline content/bn/docs/concepts/extend-kubernetes/_index.md:263]: ডেপ্রিকেটেড

File: content/bn/docs/concepts/extend-kubernetes/_index.md [hunk 8]
```diff
@@ -0,0 +1,332 @@
...
+
+FlexVolume প্লাগইনগুলো ব্যবহারকারীদের ভলিউম প্রকারগুলো মাউন্ট করার অনুমতি দেয় যা সাধারণত কুবারনেটিস দ্বারা সাপোর্টেড নয়।
+আপনি যখন FlexVolume স্টোরেজের উপর নির্ভর করে এমন একটি পড চালান, তখন kubelet ভলিউম মাউন্ট করার জন্য একটি বাইনারি প্লাগইন কল করে।
+আর্কাইভ করা [FlexVolume](https://git.k8s.io/design-proposals-archive/storage/flexvolume-deployment.md) 
+ডিজাইন প্রস্তাবে এই পদ্ধতির আরও বিশদ বিবরণ রয়েছে।
+
+[The Kubernetes Volume Plugin FAQ for Storage Vendors](https://github.com/kubernetes/community/blob/master/sig-storage/volume-plugin-faq.md#kubernetes-volume-plugin-faq-for-storage-vendors) তে
```
Comment (sajjadrahman56) [inline content/bn/docs/concepts/extend-kubernetes/_index.md:270]: স্টোরেজ বিক্রেতাদের জন্য কুবারনেটিস ভলিউম প্লাগইন FAQ

File: content/bn/docs/concepts/extend-kubernetes/_index.md [hunk 9]
```diff
@@ -0,0 +1,332 @@
...
+তাদের কাজের পরিবেশের প্রয়োজনের সাথে কীভাবে মানিয়ে নিতে হয় তা বুঝতে চায়। 
+ডেভেলপাররা যারা সম্ভাব্য {{< glossary_tooltip text="প্ল্যাটফর্ম ডেভেলপকারী" term_id="platform-developer" >}} 
+বা কুবারনেটিস প্রজেক্ট {{< glossary_tooltip text="কন্ট্রিবিউটরা" term_id="contributor" >}} , 
+এক্সটেনশন পয়েন্ট (extension points) কি এবং প্যাটার্ন বিদ্যমান এর পরিচিতি হিসাবে এবং 
+তাদের ট্রেড-অফ আর সীমাবদ্ধতা জানার জন্য  এই নির্দেশিকাটিকে দরকারী হিসেবে পাবে।
+
+কাস্টমাইজেশন পন্থাগুলিকে বিস্তৃতভাবে [কনফিগারেশনে](#কনফিগারেশন) বিভক্ত করা যেতে পারে,
+যার মধ্যে শুধুমাত্র কমান্ড লাইন আর্গুমেন্ট, লোকাল কনফিগারেশন ফাইল বা API রিসোর্স পরিবর্তন করা জড়িত;
+এবং [এক্সটেনশন](#এক্সটেনশন), যার মধ্যে অতিরিক্ত প্রোগ্রাম চালানো, অতিরিক্ত নেটওয়ার্ক সার্ভিস বা উভয়ই জড়িত।
+এই ডকুমেন্টটি মূলত এক্সটেনশন সম্পর্কে।
```
Comment (sajjadrahman56) [inline content/bn/docs/concepts/extend-kubernetes/_index.md:29]: _extensions_ -> _এক্সটেনশন_

File: content/bn/docs/concepts/extend-kubernetes/_index.md [hunk 10]
```diff
@@ -0,0 +1,332 @@
...
+
+কমান্ড আর্গুমেন্ট এবং কনফিগারেশন ফাইল সবসময় একটি হোস্ট করা কুবারনেটিস সার্ভিস বা পরিচালিত ইনস্টলেশনের 
+সাথে একটি ডিস্ট্রিবিউশন এ পরিবর্তনযোগ্য নাও হতে পারে। যখন তারা পরিবর্তনযোগ্য হয়, তারা সাধারণত ক্লাস্টার 
+অপারেটর দ্বারা পরিবর্তনযোগ্য হয়। এছাড়াও, এগুলো ভবিষ্যতের কুবারনেটিস সংস্করণে পরিবর্তন  হতে পারে, এবং সেগুলো 
+সেট করার জন্য রিস্টারটিং প্রক্রিয়া প্রয়োজন হতে পারে। এই কারণে, সেগুলি শুধুমাত্র তখনই ব্যবহার করা উচিত 
+যখন অন্য কোন বিকল্প থাকে না।
+
+বিল্ট-ইন *পলিসি API* গুলো, যেমন [ResourceQuota](/bn/docs/concepts/policy/resource-quotas/), 
+[NetworkPolicy](/bn/docs/concepts/services-networking/network-policies/) এবং Role-based Access Control 
+([RBAC](/bn/docs/reference/access-authn-authz/rbac/)), হলো বিল্ট-ইন কুবারনেটিস API যা ঘোষণামূলকভাবে কনফিগার করা নীতি সেটিংস প্রদান করে।
```
Comment (sajjadrahman56) [inline content/bn/docs/concepts/extend-kubernetes/_index.md:52]: নীতি -> পলিসি

**Review by sajjadrahman56 — COMMENTED** [review]
You could change it now or later.
I'll check it again.

File: content/bn/docs/concepts/extend-kubernetes/_index.md (hunk 8 above)
Comment (asem-hamid) [reply content/bn/docs/concepts/extend-kubernetes/_index.md:270]: nah ata localize hobe nah ata website ar baire

File: content/bn/docs/concepts/extend-kubernetes/_index.md (hunk 9 above)
Comment (asem-hamid) [reply content/bn/docs/concepts/extend-kubernetes/_index.md:29]: ???

**Comment by asem-hamid** [comment]
@sajjadrahman56 check

File: content/bn/docs/concepts/extend-kubernetes/_index.md (hunk 9 above)
Comment (sajjadrahman56) [reply content/bn/docs/concepts/extend-kubernetes/_index.md:29]: they used '_' before and after the word

Comment (asem-hamid) [reply content/bn/docs/concepts/extend-kubernetes/_index.md:29]: oh w8

**Comment by asem-hamid** [comment]
@sajjadrahman56  check now

File: content/bn/docs/concepts/extend-kubernetes/_index.md [hunk 11]
```diff
@@ -0,0 +1,332 @@
...
+kubectl দ্বারা ([প্লাগইনগুলোর সাথে প্রসারিত kubectl](/bn/docs/tasks/extend-kubectl/kubectl-plugins/) দেখুন)।
+
+### এক্সটেনশন পয়েন্ট
+
+এই ডায়াগ্রামটি একটি কুবারনেটিস ক্লাস্টারের এক্সটেনশন পয়েন্ট এবং এটি 
+অ্যাক্সেসকারী ক্লায়েন্টদের দেখায়।
+
+<!-- image source: https://docs.google.com/drawings/d/1k2YdJgNTtNfW7_A8moIIkij-DmVgEhNrn3y2OODwqQQ/view -->
+
+{{< figure src="/docs/concepts/extend-kubernetes/extension-points.png"
```
Comment (sftim) [inline content/bn/docs/concepts/extend-kubernetes/_index.md:106]: If you want, you can copy https://docs.google.com/drawings/d/1k2YdJgNTtNfW7_A8moIIkij-DmVgEhNrn3y2OODwqQQ/view and make a Bengali version of this image.

File: content/bn/docs/concepts/extend-kubernetes/_index.md [hunk 12]
```diff
@@ -0,0 +1,332 @@
...
+#### এক্সটেনশন পয়েন্ট চয়েস ফ্লোচার্ট {#এক্সটেনশন-ফ্লোচার্ট}
+
+আপনি কোথা থেকে শুরু করবেন তা নিশ্চিত না হলে, এই ফ্লোচার্টটি সাহায্য করতে পারবে৷ 
+মনে রাখবেন কিছু সমাধানে বিভিন্ন ধরনের এক্সটেনশন জড়িত থাকতে পারে।
+
+<!-- image source for flowchart: https://docs.google.com/drawings/d/1sdviU6lDz4BpnzJNHfNpQrqI9F19QZ07KnhnxVrp2yg/edit -->
+
+{{< figure src="/docs/concepts/extend-kubernetes/flowchart.svg"
+    alt="প্রয়োগকারীদের জন্য ব্যবহারের ক্ষেত্র এবং নির্দেশিকা সম্পর্কে প্রশ্ন সহ ফ্লোচার্ট। সবুজ বৃত্ত হ্যাঁ নির্দেশ করে; লাল বৃত্ত না নির্দেশ করে।"
+    class="diagram-large" caption="একটি এক্সটেনশন পদ্ধতি নির্বাচন করতে ফ্লোচার্ট গাইড" >}}
```
Comment (sftim) [inline content/bn/docs/concepts/extend-kubernetes/_index.md:157]: If you want, you can copy https://docs.google.com/drawings/d/1sdviU6lDz4BpnzJNHfNpQrqI9F19QZ07KnhnxVrp2yg/edit and make a Bengali version of this image.

File: content/bn/docs/concepts/extend-kubernetes/_index.md [hunk 13]
```diff
@@ -0,0 +1,332 @@
...
+
+ক্লায়েন্ট অনুরোধ করার জন্য একটি ব্যবহারকারীর নামের সমস্ত অনুরোধে 
+[অথেন্টিকেশন](/bn/docs/reference/access-authn-authz/authentication/)  শিরোনাম বা সার্টিফিকেট যুক্ত করে।
+
+কুবারনেটিস এর বেশ কয়েকটি বিল্ট-ইন অথেন্টিকেশন পদ্ধতি রয়েছে যা এটি সাপোর্ট করে। 
+এটি একটি অথেন্টিকেটিং প্রক্সির পিছনেও বসতে পারে এবং এটি একটি `অনুমোদন(Authorization)` টোকেন পাঠাতে পারে যাচাইয়ের জন্য: 
+শিরোনাম থেকে  একটি রিমোট সার্ভিসে (একটি [অথেন্টিকেশন webhook](/bn/docs/reference/access-authn-authz/authentication/#webhook-token-authentication)) 
+যদি সেগুলো আপনার প্রয়োজনগুলো পূরণ না করে৷
+
+### অনুমোদন(Authorization)
```
Comment (sftim) [inline content/bn/docs/concepts/extend-kubernetes/_index.md:224]:
```suggestion
### অনুমোদন(Authorization) {#authorization}
```
?

File: content/bn/docs/concepts/extend-kubernetes/_index.md (hunk 11 above)
Comment (asem-hamid) [reply content/bn/docs/concepts/extend-kubernetes/_index.md:106]: done


https://docs.google.com/drawings/d/1nZtQmJpqsOnhNagfWOMWX9DEsJyOHyvDIgTNQ9dOyC0/edit?usp=sharing

Comment (asem-hamid) [reply content/bn/docs/concepts/extend-kubernetes/_index.md:106]: done 

https://docs.google.com/drawings/d/1nZtQmJpqsOnhNagfWOMWX9DEsJyOHyvDIgTNQ9dOyC0/edit?usp=sharing

Comment (asem-hamid) [reply content/bn/docs/concepts/extend-kubernetes/_index.md:106]: https://docs.google.com/drawings/d/1nZtQmJpqsOnhNagfWOMWX9DEsJyOHyvDIgTNQ9dOyC0/view

File: content/bn/docs/concepts/extend-kubernetes/_index.md (hunk 12 above)
Comment (asem-hamid) [reply content/bn/docs/concepts/extend-kubernetes/_index.md:157]: I think this link is outdated.

image

Comment (asem-hamid) [reply content/bn/docs/concepts/extend-kubernetes/_index.md]: done 

image

File: content/bn/docs/concepts/extend-kubernetes/_index.md (hunk 13 above)
Comment (asem-hamid) [reply content/bn/docs/concepts/extend-kubernetes/_index.md:224]: Just to confirm, {#authorization} isn't in the en file. Should I add it in bn ?

image

**Review by mitul3737 — APPROVED** [review]

**Pull request merged**